site/docs/tutorials/snake/moving-the-snake.md

```diff
@@ -0,0 +1,202 @@
+---
+sidebar_label: Moving the snake
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Moving the snake
+
+To move the snake, you have to change the X and Y position of every body-part of the snake.
+
+For this, you can start at the end of the body and give this part the values of the part ahead.
+Since the final piece to get's it's values "stolen" is the head, the head seems to disappear for very brief moment.
+
+After this is done, you simply move the head in the direction of the snake.
+Since all of this happens, before the snake is rendered, it appears as a fluid motion.
+
+This is how the logic looks like:
+
+![](images/snake_move_logic.webp)
+
+Keep in mind: This is not what the player sees. This is:
+
+![](images/snake_move_rendered.webp)
+
+
+## Moving the body
+
+<Tabs
+    groupId="code-language"
+    defaultValue="language-typescript"
+    values={[
+        {label: 'AssemblyScript', value: 'language-typescript'},
+        {label: 'C / C++', value: 'language-cpp'},
+        {label: 'Rust', value: 'language-rust'},
+        {label: 'Go', value: 'language-go'},
+    ]}>
+
+<TabItem value="language-typescript">
```

Comment:
Is the idea that these other languages will be filled out later? Right now it looks a bit broken when Go is not selected.

Maybe we should remove the language pickers for now, and explain that this tutorial is geared towards Go?

Reply:
I'm currently adding AssemblyScript.
Afterwards, I try my hands on Rust.
Then I try to get C working.
And I want them all finished before I'd remove the "Draft" status.